### PR TITLE
feat(operator): expose host route in session status

### LINKED ISSUE #242: Expose external host name in Session.status

### PR DESCRIPTION
**Pull request by aslakknutsen**
Allow a client to know which session routes have been exposed for this
session.

fixes #242

### CODE REVIEW

File: pkg/istio/gateway_test.go [hunk 1]
```diff
@@ -91,6 +91,17 @@ var _ = Describe("Operations for istio gateway kind", func() {
...
 
+			It("add single session - verify ref", func() {
+				err := istio.GatewayMutator(ctx, ref)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(ref.ResourceStatuses).To(HaveLen(1))
+				Expect(ref.ResourceStatuses[0].Name).To(Equal("gateway"))
+				Expect(ref.ResourceStatuses[0].Kind).To(Equal("Gateway"))
+				Expect(ref.ResourceStatuses[0].Prop).ToNot(BeEmpty())
+				Expect(ref.ResourceStatuses[0].Prop["hosts"]).To(Equal("test.domain.com"))
```
Comment (bartoszmajsak) [inline pkg/istio/gateway_test.go:102]: Isn't `Expect(ref.ResourceStatuses[0].Prop["hosts"]).To(Equal("test.domain.com"))` enough? 101 seems somewhat redundant.

Comment (aslakknutsen) [reply pkg/istio/gateway_test.go:102]: technically...

But the error will be "map is empty" as opposed to " '' doesn't match "

Comment (aslakknutsen) [reply pkg/istio/gateway_test.go:102]: then again if more variables are added it would fail the Empty() check..

Comment (aslakknutsen) [reply pkg/istio/gateway_test.go:101]: https://github.com/maistra/istio-workspace/pull/483/commits/35f1d9e9b5c6ec5ce165e824f0f43eabfab25b2c

**Pull request merged**